src/Sulu/Bundle/TestBundle/Testing/TestUserProvider.php

```diff
@@ -58,27 +61,34 @@ class TestUserProvider implements UserProviderInterface
      */
     private $userProvider;
 
+    /**
+     * @var RoleRepositoryInterface
+     */
+    private $roleRepository;
+
     public function __construct(
         EntityManager $entityManager,
         ContactRepositoryInterface $contactRepository,
         UserRepositoryInterface $userRepository,
         EncoderFactoryInterface $userPasswordEncoderFactory,
-        UserProviderInterface $userProvider
+        UserProviderInterface $userProvider,
+        RoleRepositoryInterface $roleRepository
     ) {
         $this->entityManager = $entityManager;
         $this->contactRepository = $contactRepository;
         $this->userRepository = $userRepository;
         $this->userPasswordEncoderFactory = $userPasswordEncoderFactory;
         $this->userProvider = $userProvider;
+        $this->roleRepository = $roleRepository;
     }
 
-    public function getUser()
+    public function getUser(string $username = self::TEST_USER_USERNAME)
     {
-        if ($this->user) {
+        if ($this->user && self::TEST_USER_USERNAME === $username) {
```

Comment:
Don't you think that this is quite confusing? This class was only supposed to load a test user, now it being "abused" to create other users as well, and just caching one of them. For now I would keep that part in the tests, and if we decide at one point that we want to outsource that part, it should probably be a different service IMO.